src/test/test-fd-util.c

```diff
@@ -155,7 +155,7 @@ static void test_fd_move_above_stdio(void) {
         new_fd = fd_move_above_stdio(new_fd);
         assert_se(new_fd >= 3);
 
-        assert_se(dup(original_stdin) == 0);
+        assert_se(fcntl(original_stdin, F_DUPFD, 3) >= 3);
```

Comment:
this should remain dup() actuall. it's one of the few cases where we actually want to see that fd 0 was closed, and thus want plain dup() to fill in fd 0, and not fd 3